packages/dds/tree/src/simple-tree/api/create.ts

```diff
@@ -0,0 +1,179 @@
+import type {
+	TreeLeafValue,
+	ImplicitFieldSchema,
+	InsertableTreeFieldFromImplicitField,
+	TreeFieldFromImplicitField,
+	FieldSchema,
+	FieldKind,
+} from "../schemaTypes.js";
+import type { Unhydrated } from "../core/index.js";
+import {
+	cursorForMapTreeNode,
+	defaultSchemaPolicy,
+	FieldKinds,
+	intoStoredSchema,
+	mapTreeFromCursor,
+	type NodeKeyManager,
+} from "../../feature-libraries/index.js";
+import { getOrCreateNodeFromFlexTreeNode, type InsertableContent } from "../proxies.js";
+import { getOrCreateMapTreeNode, isFieldInSchema } from "../../feature-libraries/index.js";
+import { toFlexSchema } from "../toFlexSchema.js";
+import { inSchemaOrThrow, mapTreeFromNodeData } from "../toMapTree.js";
+import {
+	applySchemaToParserOptions,
+	cursorFromVerbose,
+	type ParseOptions,
+	type VerboseTree,
+	type VerboseTreeNode,
+} from "./verboseTree.js";
+
+/**
+ * Construct tree content compatible with a field defined by the provided `schema`.
+ * @param schema - The schema for what to construct. As this is an {@link ImplicitFieldSchema}, a {@link FieldSchema}, {@link TreeNodeSchema} or {@link AllowedTypes} array can be provided.
+ * @param data - The data used to construct the field content.
+ * @remarks
+ * When providing a {@link TreeNodeSchemaClass}, this is the same as invoking its constructor except that an unhydrated node can also be provided.
+ * This function exists as a generalization that can be used in other cases as well,
+ * such as when `undefined` might be allowed (for an optional field), or when the type should be inferred from the data when more than one type is possible.
+ *
+ * Like with {@link TreeNodeSchemaClass}'s constructor, its an error to provide an existing node to this API.
+ * TODO: For that case, use we should provide `Tree.clone`.
+ * @privateRemarks
+ * This could be exposed as a public `Tree.create` function.
+ */
+export function createFromInsertable<TSchema extends ImplicitFieldSchema>(
+	schema: TSchema,
+	data: InsertableTreeFieldFromImplicitField<TSchema>,
+	context?: NodeKeyManager | undefined,
+): Unhydrated<TreeFieldFromImplicitField<TSchema>> {
+	const cursor = cursorFromInsertable(schema, data, context);
+	const result = cursor === undefined ? undefined : createFromCursor(schema, cursor);
+	return result as Unhydrated<TreeFieldFromImplicitField<TSchema>>;
+}
+
+/**
+ * Construct tree content compatible with a field defined by the provided `schema`.
+ * @param schema - The schema for what to construct. As this is an {@link ImplicitFieldSchema}, a {@link FieldSchema}, {@link TreeNodeSchema} or {@link AllowedTypes} array can be provided.
+ * @param data - The data used to construct the field content.
+ * @remarks
+ * When providing a {@link TreeNodeSchemaClass},
+ * this is the same as invoking its constructor except that an unhydrated node can also be provided and the returned value is a cursor.
+ * When `undefined` is provided (for an optional field), `undefined` is returned.
```

Comment:
Does this apply to `identifier` fields as well? Should this say "for fields with defaults"?

Reply:
(Same with the docs for the previous method, if so)

Reply:
mapTreeFromNodeData has several issues in the identifier case. It has several TODOs and a couple bugs ( AB#9126 AB#9131 ) regarding the fact that we never validation anything even when opting in to validation which is the only reason identifier fields using defaults don't fail validation, and an additional TODO about not supporting defaults for the root case.

Even if it supported defaults for the root case, if you didn't provide the optional NodeKeyManager, it would just return undefined for an identifier and fail schema validation.

As those are all existing issue with production code this calls, and are tracked by several bugs and TODOs, I think not calling out those issues here might be ok (but we really need to get them fixed sometime soon!)